liquid-rust-typeck/src/type_env.rs

```diff
@@ -350,28 +354,30 @@ impl TypeEnvInfer {
                     .zip(sorts)
                     .map(|(e, sort)| {
                         let has_free_vars = !scope.contains_all(e.vars());
-                        if has_free_vars {
+                        if has_free_vars && !names.contains_key(e) {
```

Comment:
Shouldn't we use the existing name if `has_free_vars && names.contains_key(e)`?

Reply:
how odd -- my code has that explicit `!names.contains_key(e)` ... wonder why it didn't show up in the PR?

Reply:
ah sorry, misunderstood. Yes let me think ...

Reply:
ok fixed. take a look!